### ISSUE TITLE
how to pickle INN model and skip repeated conversion cost

### ISSUE THREAD

**Issue by JBartholomewMN** [TheStageAI/TorchIntegral#28]
I have a pre-trained resnet-50 network and I would like to experiment with different amounts of compression via this discrete -> continuous -> resample process.

What I am noticing is that it takes quite some time to construct the integral wrapper. From there it appears that resizing the model groups can only be done once. After that, any calls to resize groups fail silently and have no impact on the model that is export via get_unparameterized_model().

What this means is that each time I wish to experiment with a different amount of compression, I must reconstruct the IntegralWrapper, which is super time consuming. I've tried to pickle the wrapped model and also deepcopy it so that I can avoid this cost, and neither of them work due to recursion limit issues. Increasing the recursion limit simply crashes the kernel. 

So this issue is essentially three fundamental questions:
1. How can resize() be made to work if called multiple times without reconstructing the wrapped model each time?
2. Can the wrapped model be pickled or saved such that we can skip the time-consuming process of converting/rearranging/permuting if it has already been done once?
3. Is there a different or better way to experiment with different compression amounts without reconstructing the wrapped model each time?

**Comment by b1n0** [comment]
1. Of course unparametrized model will not be changed by another call of `group.reisze(new_size)`. Resize only changes the integral (wrapped) model, so after resize you need to call `get_unparametrized_model` again. In case when you want to measure only the accuracy of the pruned model it is enough to call `group.resize(new_size)` and use integral model in `eval` mode, but if you want to measure speed up and compression of the pruned model you should call `get_unparametrized_model` after pruning.
2. 
```
discrete_model = resnet50(pretrained=True)
wrapper_1 = IntegralWrapper(init_from_discrete=True)
integral_model = wrapper(discrete_model, example_input)
torch.save(integral_model.state_dict(), 'name.pth')

integral_model_2 = IntegralWrapper(init_from_discrete=False)(discrete_model, example_input)
integral_model_2.load_state_dict(torch.load('name.pth'))
```
3. See point 1.
